code/modules/spells/spell_types/barnyard.dm

```diff
@@ -23,7 +23,11 @@
 
 	var/mob/living/carbon/target = targets[1]
 
-	if(!(target.type in compatible_mobs))
```

Comment:
it checked the specific type but not sub types which messed it up, or something like that